src/logging.cpp

```diff
@@ -443,3 +487,22 @@ void BCLog::Logger::ShrinkDebugFile()
     else if (file != nullptr)
         fclose(file);
 }
+
+bool BCLog::Logger::SetLogLevel(const std::string& level_str)
+{
+    const auto level = GetLogLevel(level_str);
```

Comment:
```suggestion
    const auto level = GetLogLevel(level_str);
    if (level > BCLog::Level::Warning) return false;
```

Suggestion ( feel free to ignore ):

Since we are unconditionally logging any logs `>=  BCLog::Level::Warning` level, should we prevent users from setting any threshold level  `> BCLog::Level::Warning`? This will prevent users from having the weird behavior of successfully setting the threshold log level to `Error` but still seeing `Warning` logs. (Same for `SetCategoryLogLevel`)

Reply:
Was thinking about this as well. Let's leave it for a follow-up as this commit is already quite large, unless others feel strongly.

Reply:
> Suggestion ( feel free to ignore ):
> 
> Since we are unconditionally logging any logs `>= BCLog::Level::Warning` level, should we prevent users from setting any threshold level `> BCLog::Level::Warning`? This will prevent users from having the weird behavior of successfully setting the threshold log level to `Error` but still seeing `Warning` logs. (Same for `SetCategoryLogLevel`)

I updated your first commit here to do this.